zappa/handler.py

```diff
@@ -325,12 +325,12 @@ def get_function_for_aws_event(self, record):
     def get_function_for_kafka_trigger(self, event):
         arn = event.get("eventSourceArn")
         # if multiple topics are in 1 trigger, need to improve
-        if 'records' in event:
-            keys = list(event.get("records").keys())
-            topic = keys[0].rsplit("-", 1)[0]
-            if arn and topic:
-                arn = f"{arn}:{topic.strip()}"
-                return self.settings.AWS_EVENT_MAPPING.get(arn)
+        records = event.get("records")
+        keys = list(records.keys()) if records else []
+        topic = keys[0].rsplit("-", 1)[0] if keys else None
```

Comment:
Can you give an example of a sample `key` here is?
Is the topic always defined in the first key?

If defined is there only 1 key?

Reply:
https://docs.aws.amazon.com/lambda/latest/dg/with-msk.html

`{
   "eventSource":"aws:kafka",
   "eventSourceArn":"arn:aws:kafka:sa-east-1:123456789012:cluster/vpc-2priv-2pub/751d2973-a626-431c-9d4e-d7975eb44dd7-2",
   "records":{
      "mytopic-0":[
         {
            "topic":"mytopic",
            "partition":0,
            "offset":15,
            "timestamp":1545084650987,
            "timestampType":"CREATE_TIME",
            "key":"abcDEFghiJKLmnoPQRstuVWXyz1234==",
            "value":"SGVsbG8sIHRoaXMgaXMgYSB0ZXN0Lg==",
            "headers":[
               {
                  "headerKey":[
                     104,
                     101,
                     97,
                     100,
                     101,
                     114,
                     86,
                     97,
                     108,
                     117,
                     101
                  ]
               }
            ]
         }
      ]
   }
}`

Reply:
It appears that `records` supports more than 1 record, and currently only 1 record is handled.
It looks like it would be better to handle each record.

Something like this?

```
records = event.get("records", None)
if records:
    for kafka_topic in record.keys():
...
```

But it looks like the code needs to be refactored to handle multiple topics.
How are other record sources handled?

Reply:
`{
    "eventSource":"aws:kafka",
    "eventSourceArn":"arn:aws:kafka:sa-east-1:123456789012:cluster/vpc-2priv-2pub/751d2973-a626-431c-9d4e-d7975eb44dd7-2",
    "records":{
        "mytopic-0":[
            {
                "topic":"mytopic",
                "partition":0,
                "offset":15,
                "timestamp":1545084650987,
                "timestampType":"CREATE_TIME",
                "key":"abcDEFghiJKLmnoPQRstuVWXyz1234==",
                "value":"SGVsbG8sIHRoaXMgaXMgYSB0ZXN0Lg==",
                "headers":Array[1]
            }
        ],
        "mytopic-1":[
            {
                "topic":"mytopic",
                "partition":1,
                "offset":15,
                "timestamp":1545084650987,
                "timestampType":"CREATE_TIME",
                "key":"abcDEFghiJKLmnoPQRstuVWXyz1234==",
                "value":"SGVsbG8sIHRoaXMgaXMgYSB0ZXN0Lg==",
                "headers":[
                    {
                        "headerKey":[
                            104,
                            101,
                            97,
                            100,
                            101,
                            114,
                            86,
                            97,
                            108,
                            117,
                            101
                        ]
                    }
                ]
            }
        ]
    }
}`
if event like this, how to handle it? could you share some opinions?